client/my-sites/domains/controller.jsx

```diff
@@ -204,6 +205,10 @@ const googleAppsWithRegistration = ( context, next ) => {
 		page( '/checkout/' + siteSlug );
 	};
 
+	if ( isGSuiteRestricted() ) {
+		next();
+	}
```

Comment:
This is really minor but since this check is used to bail out we could move it at the beginning of the method as there is no need to initialize all these event handlers if G Suite is not available.